Wikipedia/Code/WMFExploreCollectionViewController.m

```diff
@@ -269,6 +273,12 @@ - (nullable NSURL *)contentURLForIndexPath:(NSIndexPath *)indexPath {
             return nil;
         }
         return [[content[indexPath.row] featuredArticlePreview] articleURL] ?: [[[content[indexPath.row] articlePreviews] firstObject] articleURL];
+    } else if ([section contentType] == WMFContentTypeOnThisDayEvent) {
```

Comment:
You would need to implement the case for on this day in `detailViewControllerForItemAtIndexPath` similar to how news is implemented (where it checks for an index path of more than two elements, section, item, then sub-item, where sub-item is the index of the item in the side scrolling list)